core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -2200,4 +2197,222 @@ public void testDocStats() throws IOException {
         assertEquals(0, docStats.getDeleted());
         assertEquals(numDocs, docStats.getCount());
     }
+
+    public void testDoubleDelivery() throws IOException {
```

Comment:
nothing it's additional ways to test it. Its mainly copied from the stuff we had in 1.x and I am happy to have this tested in multiple ways